Dfe.Academies.External.Web/CustomValidators/ConfirmTrueAttribute.cs

```diff
@@ -0,0 +1,46 @@
+﻿//// Example here:-
+//// https://stackoverflow.com/questions/36566836/asp-net-core-mvc-client-side-validation-for-custom-attribute
+
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using System.ComponentModel.DataAnnotations;
+
+namespace Dfe.Academies.External.Web.CustomValidators;
+
+[AttributeUsage(AttributeTargets.Property)]
+public sealed class ConfirmTrueAttribute : ValidationAttribute, IClientModelValidator
+{
+	protected override ValidationResult IsValid(object? value, ValidationContext validationContext)
+	{
+		if (!Convert.ToBoolean(value))
+			return new ValidationResult(ErrorMessage);
+		else
+			return ValidationResult.Success!;
+	}
+
+	// below inherited from IClientValidatable ??
+	////public IEnumerable<ModelClientValidationRule> GetClientValidationRules(ModelMetadata metadata, ControllerContext context)
+	////{
+	////	ModelClientValidationRule mvr = new ModelClientValidationRule(errorMessage: ErrorMessage, validationType: "confirmselection");
+	////	return new[] { mvr };
+	////}
+
+	public void AddValidation(ClientModelValidationContext context)
+	{
+		MergeAttribute(context.Attributes, "data-val", "true");
+		var errorMessage = FormatErrorMessage(context.ModelMetadata.GetDisplayName());
+		MergeAttribute(context.Attributes, "data-val-confirmselection", errorMessage);
```

Comment:
What does this line do? Do we need it and, if so, can we make it more generic

Reply:
it's to do with re-writing the HTML element. This whams in this string into the control within the HTML rendered page. I'd rather not change though as this will affect JS and might take me a while to fix / test JS